lib/simplecov.rb

```diff
@@ -342,7 +342,7 @@ def start_coverage_measurement
       if coverage_start_arguments_supported?
         start_coverage_with_criteria
       else
-        Coverage.start
+        Coverage.start unless Coverage.running?
```

Comment:
just to note, `Coverage.running?` is not implemented on Truffleruby, so this fails. I have opened an issue there: https://github.com/oracle/truffleruby/issues/2813

Reply:
Are these `unless Coverage.running?` checks necessary?
Otherwise I would suggest to drop them to avoid breaking existing TruffleRuby releases and potentially many CI runs on TruffleRuby.

Of course we'll implement the method on TruffleRuby master, but unlikely before 2023 and won't affect existing releases.

Reply:
Looks like the issue is partially described in https://github.com/simplecov-ruby/simplecov/pull/1035/commits/1cc7650454fc7c53e7a724941662c3e9d3329ccc

`Coverage.start` raises now if already running.

It also feels like something would be wrong if you started coverage while it's already running. I have no doubt though that maybe the SimpleCov code, but most definitely various degrees of client code would break.

Since we are (still :P) 0.x and "well actually" Ruby 3 broke compatibility  I'm kinda ok with that personally.

@mame am I missing something? :thinking: 

Also, hi all :wave: Been a long time, wanted to get back into this sooner but first stress, then covid hit, then bunny problems hit. Slowly but surely getting here.

Reply:
Thanks for the context. I think then the best would be to add a private method like `coverage_enabled?` and track it as an ivar on `SimpleCov` and use `Coverage.running?` if available and that ivar otherwise. I'll try to make a PR.

Reply:
I opened a PR somewhat like that: https://github.com/simplecov-ruby/simplecov/pull/1043 - I'm not sure the ivar is needed

I hope your bunny(ies) are well, @PragTob.